core/src/main/scala/aima/core/agent/Actuator.scala

```diff
@@ -1,19 +1,23 @@
 package aima.core.agent
 
-import aima.core.random.Randomness
+import aima.core.random.{DefaultRandomness, Randomness}
 
 /**
   * @author Shawn Garner
   * @author Damien Favre
   */
 trait Actuator[ENVIRONMENT, ACTION] {
-  def act(action: ACTION, e: ENVIRONMENT): ENVIRONMENT //could be Action => E => E
+  def act(action: ACTION, e: ENVIRONMENT): Option[ENVIRONMENT]
```

Comment:
I'm not sure I like the idea of returning an option of an environment. 
I was thinking of the environment as immutable and this as like a state monad where if there is no action taken then it's basically identity otherwise it applies the function and gives you a new immutable state.

Reply:
Yeah I agree with you. I think the Actuator act method returning an Option isn't a good idea after all. No matter if the actuators work reliably or not, you'll always get an Environment 👍 I'll fix it.